mono/mini/mini.c

```diff
@@ -3685,6 +3685,8 @@ mini_method_compile (MonoMethod *method, guint32 opts, MonoDomain *domain, JitFl
 		mono_local_deadce (cfg);
 	if (cfg->opt & MONO_OPT_ALIAS_ANALYSIS)
 		mono_local_alias_analysis (cfg);
+	if (cfg->opt & MONO_OPT_DIVREM && cfg->has_divisions)
+		mono_local_div_imm_removal (cfg);
```

Comment:
I think having a separate pass is fine. Simple passes are incredibly fast.

If emulation is a problem, what we could do, which might actually be a good idea, is to turn emulation into a pass, always emit OPs in the front end and do the same trick you did with cfg->has_divisions.